chingu-voyages/v52-tier3-team-38
# Dynamic header! Dynamic footer! Very demure!

# Description

This pull request refactors header and navbar components to support multiple user roles (unauthenticated users, authenticated users, and admins) and improves error handling in the RootLayout component. These changes streamline user authentication and enhance code organization for better maintainability.

## Related to the following Jira/Trello issue 

- Header card
- Navigation Bar card

## Main changes explained:

- Add new components:

 -- Created AdminHeader and AdminNavbar components to support admin-specific navigation and headers.
-- Refactored UserHeader and UserNavbar components for non-admin users.

- Refactor existing layout:

-- Updated RootLayout to fetch user details dynamically and conditionally render the appropriate headers and navbars based on user role.
-- Removed unused font imports to clean up the codebase.
-- Improved error handling for user fetching, ensuring a better user experience during authentication issues.

- Fix bugs and improve usability:

-- Corrected casing in UserHeader role messages for consistency.
-- Fixed conditional logic in RootLayout to properly check for user roles using integer values (e.g., 1 instead of "1").

## How to test:
1. Check into Ross/feature/dynamic-header-footer
2. Do `npm install` and `npm run dev` commands to run this PR locally
3.  You should be able to look at the header and footer nav. 
4. Log out if necessary, to see the unauthenticated version.
5. Log in as Bruce Wayne to see the admin header and admin footer nav.

## Screenshots or videos of changes:

![image](https://github.com/user-attachments/assets/1bdd833f-06a3-493e-98d2-4697bf6e0ec3)

![image](https://github.com/user-attachments/assets/b4104819-51f4-4b71-bb17-cdd78a90e868)

![image](https://github.com/user-attachments/assets/9a9b9692-66ff-4df2-a39f-f73b5c385487)

![image](https://github.com/user-attachments/assets/efbe449b-8930-4de8-8da0-597f14ea461d)



## Note:
Include any information the reviewers need to know.
